### PR TITLE
[TrilinosApplication] Adding additional checks and memory optimizations into sparse matrices in block B&S

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Adding additional checks and memory optimizations (basically calling `OptimizeStorage` method) into sparse matrices in block B&S.

Also using some simple C arrays instead of std::vector to simplify allocations.

**🆕 Changelog**

- [Adding additional checks and memory optimizations into sparse matrices in block B&S](https://github.com/KratosMultiphysics/Kratos/commit/a509910658617dab941b2b6c01d1a036aa4da7c6)

### CODE REVIEW

File: applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h [hunk 1]
```diff
@@ -1225,10 +1229,10 @@ class TrilinosBlockBuilderAndSolver
 
             // Adding diagonal values
             int ierr;
-            std::vector<int> index_diagonal(1, 0);
+            int index_diagonal[1];
```
Comment (roigcarlo) [inline applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h]:
```suggestion
            int index_diagonal[1] = {0};
```

File: applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h [hunk 2]
```diff
@@ -1257,18 +1261,36 @@ class TrilinosBlockBuilderAndSolver
 
                 // Adding cross master-slave dofs
                 if (num_active_slave_indices > 0 && num_active_master_indices > 0) {
-                    std::vector<int> slave_index(1, 0);
+                    int slave_index[1];
```
Comment (roigcarlo) [inline applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h]:
```suggestion
                    int slave_index[1] = {0};
```

File: applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h [hunk 3]
```diff
@@ -1544,10 +1550,10 @@ class TrilinosBlockBuilderAndSolver
 
             // First adding the pure slave dofs
             if (num_active_slave_indices > 0) {
-                std::vector<int> index(1, 0);
+                int index[1];
```
Comment (roigcarlo) [inline applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h]:
```suggestion
                int index[1] = {0};
```

File: applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h [hunk 4]
```diff
@@ -1558,10 +1564,10 @@ class TrilinosBlockBuilderAndSolver
             }
             // Second adding pure master dofs
             if (num_active_master_indices > 0) {
-                std::vector<int> index(1, 0);
+                int index[1];
```
Comment (roigcarlo) [inline applications/TrilinosApplication/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h]:
```suggestion
                int index[1] = {0};
```

**Comment by loumalouomega** [comment]
Suggestion @roigcarlo, add code suggestions as comment, not as review, so I can join suggestions

**Comment by roigcarlo** [comment]
Okkk

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**